analytic_engine/src/table/data.rs

```diff
@@ -69,12 +75,108 @@ pub enum Error {
     FindMemTable {
         source: crate::table::version::Error,
     },
+
+    #[snafu(display("Failed to persist max file id, err:{}", source))]
+    PersistFileId { source: GenericError },
 }
 
 define_result!(Error);
 
 pub type MemTableId = u64;
 
+const ALLOC_STEP: u64 = 1000;
+
+#[derive(Debug)]
+pub struct Allocator {
+    last_file_id: FileId,
+    max_file_id: FileId,
+    alloc_step: u64,
+}
+
+impl Default for Allocator {
+    fn default() -> Self {
+        Self {
+            last_file_id: 0,
+            max_file_id: 0,
+            alloc_step: ALLOC_STEP,
+        }
+    }
+}
+
+impl Allocator {
+    ///New a allocator
+    pub fn new(last_file_id: FileId, max_file_id: FileId) -> Self {
+        Self {
+            last_file_id,
+            max_file_id,
+            alloc_step: ALLOC_STEP,
+        }
+    }
+
+    /// Returns the last file id
+    fn last_file_id(&self) -> FileId {
+        self.last_file_id
+    }
+
+    /// Set max file id
+    fn set_max_file_id(&mut self, max_file_id: FileId) {
+        self.max_file_id = max_file_id;
+    }
+
+    /// Returns the max file id
+    fn max_file_id(&self) -> FileId {
+        self.max_file_id
+    }
+
+    /// Return the alloc step
+    fn alloc_step(&self) -> u64 {
+        self.alloc_step
+    }
+
+    /// Determine whether the ID is used up
+    fn is_exhausted(&self) -> bool {
+        self.last_file_id() >= self.max_file_id()
+    }
+
+    /// Alloc file id
+    async fn alloc_file_id(
+        &mut self,
+        manifest: &ManifestRef,
```

Comment:
How about passing a closure called `persist_next_max_id` if we put the IdAllocator as a separate library in the `common_util`.